lib/compare.js

```diff
@@ -95,8 +95,26 @@ function errorCompare(err1, err2) {
   );
 }
 
+const removeCirculars = (obj, path, paths, compareSymbol) =>
+  typeof obj === 'object' && obj
+    ? Object.entries(obj).map(([key, value]) => [
+        key,
+        paths.set(obj, [path, compareSymbol]).get(value) ||
```

Comment:
This always `set`s with the same key `obj`, is that ok?

Reply:
Shouldn't this whole set be reversed? First you check with `get` and only then use `set` and `removeCirculars`?